### PR TITLE
Fixed api_doc_gen duplicated entries

### LINKED ISSUE #67: API documentation for nonexistent routes generated when argspec and non-argspec method present in same handler
Examples:
* https://github.com/hfaran/UBCRECSessions/blob/master/docs/API_Documentation.md#apisessionsessionpsession_ida-za-z0-9_
* https://github.com/hfaran/UBCRECSessions/blob/master/docs/API_Documentation.md#apisessionsession

Both of those have four methods listed when they shouldn't.

### PR DESCRIPTION
**Pull request by mauler**
Resolves #67

### CODE REVIEW

**Comment by hfaran** [comment]
Hey @mauler thanks for all your incredible work over the past couple weeks! This is all quite impressive.
As I mentioned earlier though, I am quite busy with school at the moment so I unfortunately do not have the spare time to look at these PRs right now, but I will certainly get them in a few weeks hopefully. I have not forgotten about them!

File: tornado_json/api_doc_gen.py [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
 import inspect
+import json
+import re
 
 try:
     from itertools import imap as map  # PY2
 except ImportError:
     pass
 
+from jsonschema import ValidationError, validate
```
Comment (hfaran) [inline tornado_json/api_doc_gen.py]: Why was this import moved up? I think it was fine where it was before.

File: tornado_json/api_doc_gen.py [hunk 2]
```diff
@@ -65,7 +67,21 @@ def _get_tuple_from_route(route):
...
+        method = getattr(handler_class, http_method, None)
+        if method:
+            method = extract_method(method)
+            method_params = set(getattr(method, "__argspec_args",
+                                        inspect.getargspec(method).args))
+            if route_params.issubset(method_params) and \
+                    method_params.issubset(route_params):
+                methods.append(http_method)
+
+    return pattern, handler_class, methods
```
Comment (hfaran) [inline tornado_json/api_doc_gen.py]: Can you change the docstring to reflect that methods are now also returned?

**Comment by hfaran** [comment]
Just the two comments I left above, otherwise :+1: 

Thanks for fixing this.

**Comment by mauler** [comment]
Fixed.
